src/rce.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * rce stands for Requested Command Execution
+ * (or Request for Command Execution if you don't mind the omitted 'f')
+ */
+
+import * as vscode from 'vscode';
+import { ActionData, ActionWithHandler } from './neuro_client_helper';
+import { NEURO } from './constants';
+import { logOutput } from './utils';
+
+/**
+ * A prompt parameter can either be a string or a function that converts ActionData into a prompt string.
+ */
+export type PromptGenerator = string | ((actionData: ActionData) => string);
+
+export const cancelRequestAction: ActionWithHandler = {
+    name: 'cancel_request',
+    description: 'Cancel the current request.',
+    permissions: [],
+    handler: handleCancelRequest,
+    promptGenerator: () => '', // No prompt needed for this action
+};
+
+/**
+ * Handles cancellation requests from Neuro.
+ */
+export function handleCancelRequest(_actionData: ActionData): string | undefined {
+    if (!NEURO.rceCallback) {
+        return 'No active request to cancel.';
+    }
+    clearRceDialog();
+    return 'Request cancelled.';
+}
+
+/**
+ * RCE's emergency shutdown component
+ * Automatically clears the RCE dialog and tell Neuro her requests was cancelled.
+ * Only runs if there is an RCE callback in NEURO.
+ */
+export function emergencyDenyRequests(): void {
+    if (!NEURO.rceCallback) {
+        return;
+    }
+    clearRceDialog();
+    logOutput("INFO", `Cancelled ${NEURO.rceCallback} due to emergency shutdown.`)
+    NEURO.client?.sendContext("Your last request was denied.")
+    vscode.window.showInformationMessage("The last request from Neuro has been denied automatically.")
+}
+
+export function clearRceDialog(): void { // Function to clear out RCE dialogs
+    NEURO.rceCallback = null;
+    NEURO.client?.unregisterActions(['cancel_request']);
```

Comment:
Reference the action instead of using a string literal to prevent accidentally changing one but not the other.
```suggestion
    NEURO.client?.unregisterActions([cancelRequestAction.name]);
```